src/app/components/Transaction/InfoBox.tsx

```diff
@@ -22,7 +22,7 @@ export function InfoBox(props: DetailProps) {
       direction="row"
       gap="small"
       hoverIndicator={{ color: 'background-contrast' }}
-      onClick={() => boxClicked()}
+      onClick={props.link ? boxClicked : undefined}
```

Comment:
Ah, nice!

I'd remove `if (props.link) {` from `boxClicked` too, but then typescript will complain.

And I'd replace onClick with wrapping NavLink around Box (real link with right-click and middle-click), but making it a conditional-wrapper results in another component.

Maybe this is better solved one level higher:
```diff
  <InfoBox icon={<Money color="brand" />} label={t('common.amount', 'Amount')} value={amount} />
+ <NavLink to={`/account/${otherAddress}`}>
    <InfoBox
      icon={<ContactInfo color="brand" />}
      label={designation}
      value={<ShortAddress address={otherAddress} />}
-     link={`/account/${otherAddress}`}    
    />
+ </NavLink>


- link?: string
- onClick={props.link ? boxClicked : undefined}
```

Reply:
looks much better, thanks! will bump PR today